Integrate Tricktrack without new externals

vvolkl opened the pull request:
Almost the same as #282, but lacking two things, which allows to merge this with current externals:
* Tracks are not associated to hits, which needs the change in edm v0.5.3
* The RiemannFit header is moved temporarily from tricktrack to FCCSW, if new versions of tricktrack are available in externals it should be removed

vvolkl commented:
@phsft-bot  build please

vvolkl commented:
@phsft-bot build please

zaborowska commented on Reconstruction/RecTracker/RecTracker/RiemannFit.h:
has been verified by whom?

zaborowska commented on Reconstruction/RecTracker/RecTracker/RiemannFit.h:
checkformat

zaborowska commented on Reconstruction/RecTracker/RecTracker/TTPoint.h:
```diff
@@ -0,0 +1,38 @@
+#include <array>
+
+namespace tricktrack {
+
```
missing doxy

zaborowska commented on Reconstruction/RecTracker/options/TrickTrackReco.py:
```diff
@@ -0,0 +1,191 @@
+parser.add_argument('--npileup', type=int, default=0, help='specify number of events to process')
+parser.add_argument('--noSignal', action="store_true",  help='use old collection names')
+parser.add_argument('--cleanHits', action="store_true",  help='use old collection names')
+parser.add_argument('--legacyCollectionNames', action="store_true",  help='use old collection names')
```
maybe you put legacyCollectionNames and overlayCollectionNames in a mutually exclusive group?

vvolkl replied:
Removed those arguments alltogether. Was there only because I had some old samples  I was doing checks on, and it saved time. However, we should maybe take a look at the branch names we are using in the examples and set some conventions - I believe in the FCC Sim Jobs this is done better (`TrackerPositionedHits` vs. `positionedHits`)

zaborowska commented on Reconstruction/RecTracker/options/TrickTrackReco_withTime.py:
```diff
@@ -0,0 +1,191 @@
+import os
```
isn't there a way to mergo both those py configs?

zaborowska commented on Reconstruction/RecTracker/scripts/fakeRate.py:
```diff
@@ -0,0 +1,390 @@
+"""
```
A comment like that would be great in the rest of configs

zaborowska commented on Reconstruction/RecTracker/src/components/FastHitFilterTool.cpp:
```diff
@@ -0,0 +1,57 @@
+#include "FastHitFilterTool.h"
+
+
+#include "datamodel/TrackHitCollection.h"
```
checkformat

zaborowska commented on Reconstruction/RecTracker/src/components/FastHitFilterTool.h:
```diff
@@ -0,0 +1,42 @@
+#include "tricktrack/HitChainMaker.h"
+
+#include <map>
+
```
missing doxy.
Also since it's a component, its name should suggest it is used for TrackHits.

zaborowska commented on Reconstruction/RecTracker/src/components/FullInnerLayerGraphTool.h:
```diff
@@ -0,0 +1,23 @@
+
+#include "tricktrack/CMGraph.h"
+
+
```
This is for use with FCChhTracker, right?
I'd suggest then to disentangle it from RecTracker and create RecFCChhTracker.
And I'd add `Tracker` to the name, maybe instead of `Full` or `Layer` (or both).

zaborowska commented on Reconstruction/RecTracker/src/components/RecHelixTrajectory.h:
```diff
@@ -0,0 +1,49 @@
+class TrackStateCollection;
+class PositionedTrackHitCollection;
+}
+
```
doxy comment

zaborowska commented on Reconstruction/RecTracker/src/components/RecParticleAssociations.h:
```diff
@@ -0,0 +1,42 @@
+class MCParticleCollection;
+class ParticleMCParticleAssociationCollection;
+}
+
```
missing comment

zaborowska commented on Reconstruction/RecTracker/src/components/RecTrackAlg.h:
```diff
@@ -0,0 +1,53 @@
+  /// TrackHits as Input to the track seeding
+  DataHandle<fcc::PositionedTrackHitCollection> m_positionedTrackHits{"TrackerPositionedHits", Gaudi::DataHandle::Reader,
+                                                                      this};
+  DataHandle<fcc::TrackCollection> m_tracks{"Tracks", Gaudi::DataHandle::Writer, this};
```
please add missing comments

zaborowska commented on Reconstruction/RecTracker/src/components/RiemannFitTool.cpp:
```diff
@@ -0,0 +1,117 @@
+#include "RecTracker/TTPoint.h"
+
+
+
```
run checkformat

zaborowska commented on Reconstruction/RecTracker/src/components/RiemannFitTool.h:
```diff
@@ -0,0 +1,42 @@
+
+#include <map>
+
+
```
doxy + checkformat

zaborowska commented on Reconstruction/RecTracker/src/components/TrickTrackSeedingTool.h:
```diff
@@ -0,0 +1,92 @@
+  class TTPoint;
+}
+
+
```
doxy

vvolkl replied:
done

zaborowska commented on Sim/SimG4Components/src/SimG4GeantinoDeposits.cpp:
```diff
@@ -3,13 +3,85 @@
+   // search in particle table
+   G4ParticleTable* pTable = G4ParticleTable::GetParticleTable();
+   G4ParticleDefinition* anInstance = pTable->FindParticle(name);
+   if (anInstance ==0)
```
`0` -> `nullptr`

vvolkl replied:
done

zaborowska commented on Sim/SimG4Components/src/SimG4GeantinoDeposits.cpp:
```diff
@@ -3,13 +3,85 @@
 #include "G4ProcessType.hh"
 
 
+ 
```
I actually forgot this exists. However, I think the name ` SimG4GeantinoDeposits` is quite misleading, what do you think? Wouldn't `SimG4GeantinoPhysicsList` be better?

vvolkl replied:
Funny, I would say it's just the other way around. `PhysicsList` suggests that it includes some physics processes, but in the end it just lets geantinos deposit energy in sensitive volumes, so I thought the name is fairly descriptive. I can change it though, if you are not convinced!

zaborowska replied:
The way I am looking is that G4 requires a user to give a physics list (of processes that may happen, no matter if it is "true" physics or "fake" physics). Therefore we either give e.g. FTFP_BERT or we give that Geantinos. I am not really sure what is the best name, I just wanted to make it more clear that is is the file user must put to "physicslist" property of SimG4Svc... We actually do have yet another list we use for testing and it is called `SimG4TestPhysicsList` and it kills particles at their origin, so kind of non-physical as well...

vvolkl replied:
I see your point, agreed!

vvolkl commented on Reconstruction/RecInterface/RecInterface/IHitFilterTool.h:
```diff
@@ -0,0 +1,30 @@
+
+// Gaudi
+#include "GaudiKernel/IAlgTool.h"
+
```
done

vvolkl commented on Reconstruction/RecInterface/RecInterface/IHitFilterTool.h:
```diff
@@ -0,0 +1,30 @@
+class TrackHit;
+}
+
+/** \brief Stateful filter of FCC hits based on their CellId.
```
Sure. Actually if it's just one line I don't think brief is needed, anyway.

vvolkl commented on Detector/DetFCChhBaseline1/compact/FCChh_DectEmptyTracker.xml:
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Moved to Tracker directory. The point of this was to quickly be able to reduce the world size for simulations of the tracker only, since this saves a lot of cpu time. So I think it makes sense to keep the convention similar  to  `FCChh_DectEmptyMaster`.

vvolkl commented on Detector/DetFCChhBaseline1/compact/FCChh_DectEmptyTracker.xml:
```diff
@@ -0,0 +1,26 @@
+    <gdmlFile  ref="../../DetCommon/compact/materials.xml"/>
+  </includes>
+
+  <info name="FCCDectMaster"
```
done

vvolkl commented:
Some comments were on `RiemannFit.h` and `TTPoint.h`, which are actually part of tricktrack, but not available in FCCSW until we get the new externals, that's why I added them here as a workaround. I added a comment that they'll be removed in the future

vvolkl commented on Reconstruction/RecTracker/RecPlotsMakefile.mk:
```diff
@@ -0,0 +1,28 @@
+###########################
+# FCC Track Reco Plots Makefile
```
I like having the bash commands in the makefile as it's easy to keep them in sync and they're testable, but it's true that RecTracker was still missing some docs. It will have to be expanded, but I pushed a start (Reconstruction/doc/RecTracker.md)

JavierCVilla commented:
@vvolkl @zaborowska How should we proceed here now that we have new externals? The new version (93.0.0) is now basically the same we had but decoupled from FCCSW, so we could even release a new 93.1.0 if deemed necessary.
